web/docs/language/features.md

```diff
@@ -584,6 +584,107 @@ import { isPrismaError, prismaErrorToHttpError } from '@wasp/utils.js'
   }
 ```
 
+## APIs
+
+In Wasp, the default client-server interaction mechanism is through Operations. However, if you need a specific URL method/path, or a specific response, Operations may not be suitable for you. For these cases, you can use an `api`! Best of all, they should look and feel very familiar.
+
+### API
+
+APIs are used to tie a JS function to an HTTP (method, path) pair. They are distinct from Operations, and have no client-side helpers (like `useQuery`).
+
+To create a Wasp API, you must:
+1. Define the APIs NodeJS implementation
+2. Declare the API in Wasp using the `api` declaration
+
+After completing these two steps, you'll be able to call the API from client code (via our Axios wrapper), or from the outside world.
+
+#### Defining the APIs NodeJS implementation
+An API should be implemented as a NodeJS function that takes three arguments.
+1. `req`:  Express Request object
+2. `res`: Express Response object
+3. `context`: An additional context object **injected into the API by Wasp**. This object contains user session information, as well as information about entities. The examples here won't use the context for simplicity purposes. You can read more about it in the [section about using entities in APIs](#using-entities-in-apis).
+
+Here's an example of a simple API:
+```ts title="src/server/apis.ts"
+import { FooBar } from '@wasp/apis/types'
+
+export const fooBar : FooBar = (req, res, context) => {
+  res.set('Access-Control-Allow-Origin', '*') // Example of modifying headers to override Wasp default CORS middleware.
+  res.json({ msg: `Hello, ${context.user?.username || "stranger"}!` })
+}
+```
+
+#### Declaring an API in Wasp
+After implementing your APIs in NodeJS, all that's left to do before using them is tell Wasp about it!
+You can easily do this with the `api` declaration, which supports the following fields:
+- `fn: ServerImport` (required) - The import statement of the APIs NodeJs implementation.
+- `httpRoute: (HttpMethod, string)` (required) - The HTTP (method, path) pair, where method can be one of:
+  - `ALL`, `GET`, `POST`, `PUT` or `DELETE`
+  - and path is an Express path `string`.
+- `entities: [Entity]` (optional) - A list of entities you wish to use inside your API.
+We'll leave this option aside for now. You can read more about it [here](#using-entities-in-apis).
+- `auth: bool` (optional) - If auth is enabled, this will default to `true` and provide a `context.user` object. If you do not wish to attempt to parse the JWT in the Authorization Header, you may set this to `false`.
```

Comment:
Now I got it, you get the `user` object if the auth is enabled. 

This was a bit confusing for me a bit because I expected that it will return `401` if the user is not authenticaed. 

For operations, we include the user object by default, right? This means we have two strategies: for operations we include it by default, for APIs it's opt in. Which is fine, since APIs are more hands on and used for custom work.

Reply:
Yeah it behaves the same as ops for now 👍🏻